coalib/parsing/FilterHelper.py

```diff
@@ -1,11 +1,17 @@
 from coalib.parsing.InvalidFilterException import InvalidFilterException
-from coalib.parsing.filters import available_filters
+from coalib.parsing.filters import (
+    available_filters, available_section_filters)
+from coalib.parsing.DefaultArgParser import default_arg_parser
 
 
 def is_valid_filter(filter):
     return filter in available_filters
 
 
+def is_valid_section_filter(filter):
```

Comment:
nope.

lets improve the filters system to support sections.

`def foo_filter(thing, args):`

every filter than checks whether is knows what is in `thing`, e.g. using isinstance , and raises NotImplementedError if it doesnt get it.  Existing filters only know about `Bear`.

You make a small chance to the existing filters to make them compliant with the new API by adding a `raise NotImplementedError` to them.

(Actually most of them understand could easily be improved to understand `Section`, but that is beyond the scope of your change, and each will need to be carefully done wrt the CLI args to ensure we build a good user experience)